src/main/java/net/dv8tion/jda/internal/handle/ChannelDeleteHandler.java

```diff
@@ -91,19 +92,26 @@ protected Long handleInternally(DataObject content)
                     return null;
                 }
 
-                // This is done in the AudioWebSocket already
-//                //We use this instead of getAudioManager(Guild) so we don't create a new instance. Efficiency!
-//                AudioManagerImpl manager = (AudioManagerImpl) getJDA().getAudioManagersView().get(guild.getIdLong());
-//                if (manager != null && manager.isConnected()
-//                        && manager.getConnectedChannel().getIdLong() == channel.getIdLong())
-//                {
-//                    manager.closeAudioConnection(ConnectionStatus.DISCONNECTED_CHANNEL_DELETED);
-//                }
+                //  This is done in the AudioWebSocket already
+                //  We use this instead of getAudioManager(Guild) so we don't create a new instance. Efficiency!
+                //  AudioManagerImpl manager = (AudioManagerImpl) getJDA().getAudioManagersView().get(guild.getIdLong());
+                //  if (manager != null && manager.isConnected()
+                //          && manager.getConnectedChannel().getIdLong() == channel.getIdLong())
+                //  {
+                //      manager.closeAudioConnection(ConnectionStatus.DISCONNECTED_CHANNEL_DELETED);
+                //  }
                 guild.getVoiceChannelsView().remove(channel.getIdLong());
                 getJDA().handleEvent(
-                    new ChannelDeleteEvent(
-                        getJDA(), responseNumber,
-                        channel));
+                        new ChannelDeleteEvent(
+                                getJDA(), responseNumber,
+                                channel));
+
+                // Deleting any guild scheduled events associated to the deleted channel as they are deleted when the channel gets deleted.
+                // There is no delete event for the deletion of guild scheduled events in this case, so we do this to keep the cache in sync.
+                String channelId1 = Long.toUnsignedString(channelId);
+                guild.getScheduledEventsView().stream()
+                        .filter(guildScheduledEvent -> guildScheduledEvent.getType().equals(GuildScheduledEvent.Type.VOICE) && guildScheduledEvent.getLocation().equals(String.valueOf(channelId1)))
```

Comment:
Move this outside the switch-case, we don't need to handle specific types since the ID is unique.

```suggestion
                        .filter(guildScheduledEvent -> guildScheduledEvent.getType().isChannel() && guildScheduledEvent.getLocation().equals(String.valueOf(channelId1)))
```